controllers/release/release_adapter_test.go

```diff
@@ -0,0 +1,421 @@
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "test-snapshot-",
+				Namespace:    "default",
+			},
+			Spec: appstudioshared.ApplicationSnapshotSpec{
+				Application: "testapplication",
+				Components:  []appstudioshared.ApplicationSnapshotComponent{},
+			},
+		}
+		Expect(k8sClient.Create(ctx, applicationSnapshot)).Should(Succeed())
+
+		// required by the test at line #274
+		applicationSnapshot.TypeMeta.Kind = "applicationSnapshot"
+		Expect(k8sClient.Update(ctx, applicationSnapshot)).Should(Succeed())
+	})
+
+	BeforeEach(func() {
+		release = &appstudiov1alpha1.Release{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "test-release-",
+				Namespace:    testNamespace,
+			},
+			Spec: appstudiov1alpha1.ReleaseSpec{
+				ApplicationSnapshot: applicationSnapshot.GetName(),
+				ReleaseLink:         releaseLink.GetName(),
+			},
+		}
+		Expect(k8sClient.Create(ctx, release)).Should(Succeed())
+
+		adapter = NewAdapter(release, ctrl.Log, k8sClient, ctx)
+		Expect(reflect.TypeOf(adapter)).To(Equal(reflect.TypeOf(&Adapter{})))
+	})
+
+	AfterEach(func() {
+		err := k8sClient.Delete(ctx, release)
+		Expect(err == nil || errors.IsNotFound(err)).To(BeTrue())
+	})
+
+	AfterAll(func() {
+		err := k8sClient.Delete(ctx, releaseLink)
+		Expect(err == nil || errors.IsNotFound(err)).To(BeTrue())
+
+		err = k8sClient.Delete(ctx, applicationSnapshot)
+		Expect(err == nil || errors.IsNotFound(err)).To(BeTrue())
+
+		Expect(k8sClient.Delete(ctx, releaseStrategy)).Should(Succeed())
+	})
+
+	It("can create a new Adapter instance", func() {
+		Expect(reflect.TypeOf(NewAdapter(release, ctrl.Log, k8sClient, ctx))).
+			To(Equal(reflect.TypeOf(&Adapter{})))
+	})
+
+	It("ensures finalizer is added to the Release object", func() {
+		result, err := adapter.EnsureFinalizerIsAdded()
+		Expect(!result.CancelRequest && err == nil).To(BeTrue())
+
+		Expect(release.ObjectMeta.GetFinalizers()[0]).
+			To(ContainSubstring("release-finalizer"))
+
+		result, err = adapter.EnsureFinalizerIsAdded()
+		Expect(!result.CancelRequest && err == nil).To(BeTrue())
+	})
+
+	It("ensures a PipelineRun object exists", func() {
+		// some racing condition in EnvTest makes it throw an error even if the CancelRequest is false
+		// so I am adding a function to wait for it to succeed before continuing.
+		Eventually(func() bool {
+			result, err := adapter.EnsureReleasePipelineRunExists()
+			return (!result.CancelRequest && err == nil)
+		}, time.Second*10).Should(BeTrue())
+
+		pipelineRun, _ := adapter.getReleasePipelineRun()
+		Expect(reflect.TypeOf(pipelineRun)).
+			To(Equal(reflect.TypeOf(&tektonv1beta1.PipelineRun{})))
+		Expect(pipelineRun).NotTo(BeNil())
+
+		// clean up
+		Expect(k8sClient.Delete(ctx, pipelineRun)).Should(Succeed())
+	})
+
+	It("ensures the Release Pipeline is tracked", func() {
+		Eventually(func() bool {
+			result, err := adapter.EnsureReleasePipelineRunExists()
+			return (!result.CancelRequest && err == nil)
```

Comment:
Redundant parenthesis:
```suggestion
			return !result.CancelRequest && err == nil
```

Reply:
Even the IDE's complain about it ;)